monkey/monkey_island/cc/setup/agent_event_handlers.py

```diff
@@ -31,7 +35,9 @@ def _subscribe_and_store_to_event_repository(container: DIContainer):
         container.resolve(ICredentialsRepository)
     )
     agent_event_queue.subscribe_type(CredentialsStolenEvent, save_stolen_credentials_subscriber)
-    agent_event_queue.subscribe_type(AgentShutdownEvent, update_agent_shutdown_status)
+    agent_event_queue.subscribe_type(
+        AgentShutdownEvent, update_agent_shutdown_status(container.resolve(IAgentRepository))
```

Comment:
Maybe extract `container.resolve(IAgentRepository)` to a variable like it's done with `agent_event_queue`